src/index.ts

```diff
@@ -59,39 +89,35 @@ const dmMachine = createMachine(
               }),
             ],
           },
-          Ready: {
-            initial: "Greeting",
+          Form: {
+            initial: "Prompt",
             states: {
-              Greeting: {
-                entry: "speak.greeting",
-                on: { SPEAK_COMPLETE: "HowCanIHelp" },
-              },
-              HowCanIHelp: {
-                entry: say("You can say whatever you like."),
+              Prompt: {
+                entry: "speak.prompt",
                 on: { SPEAK_COMPLETE: "Ask" },
               },
               Ask: {
                 entry: listen(),
                 on: {
                   RECOGNISED: {
-                    target: "Repeat",
+                    target: "ColourSlot",
+                    guard: ({ context }) => !!getEntity(context, "colour"),
                     actions: [
-                      ({ event }) => console.log(event),
                       assign({
-                        lastResult: ({ event }) => event.value,
```

Comment:
this way there is no lastResult anymore. You need to either assign it from the RECOGNISED event's value or just use the event directly, i.e. `({event}) => !!(grammar[event.value[0].utterance] || {}).colour` (perhaps not exactly this line, it depends on the structure of your grammar). 
Note: `|| {}` allows `grammar[event.value[0].utterance]` to be `undefined`

Reply:
It finally works! Thank you so much :)